inference-engine/src/cldnn_engine/ops/tensor_iterator.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cldnn_program.h"
+#include "cldnn_common_utils.h"
+#include "cldnn_engine.h"
+
+#include <cpp/ie_cnn_network.h>
+
+#include "ngraph/op/tensor_iterator.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/util/sub_graph_base.hpp"
+
+#include "api/loop.hpp"
+#include "api/mutable_data.hpp"
+#include "api/reorder.hpp"
+
+#include <vector>
+#include <algorithm>
+
+using TensorIterator = ngraph::op::v0::TensorIterator;
+
+namespace CLDNNPlugin {
+
+static cldnn::mutable_data CreateIntData(Program &p, const cldnn::primitive_id& id, int32_t num) {
```

Comment:
How about CreateScalarData?

Reply:
In loop primitive, there is no case to create the data whose element is floating point number. If you want to express there is the only one element in the function name, how about using `CreateSingleIntData`()